ui/page/transaction/transactions_page.go

```diff
@@ -209,99 +276,159 @@ func (pg *TransactionsPage) loadTransactions(offset, pageSize int32) ([]*sharedW
 	if err != nil {
 		err = fmt.Errorf("Error loading transactions: %v", err)
 	}
-	return tempTxs, len(tempTxs), isReset, err
+
+	txs := make([]*multiWalletTx, 0)
+	for _, tx := range tempTxs {
+		txs = append(txs, &multiWalletTx{tx, wal.GetWalletID()})
+	}
+
+	return txs, len(txs), isReset, err
 }
 
 // Layout draws the page UI components into the provided layout context
 // to be eventually drawn on screen.
 // Part of the load.Page interface.
-func (pg *TransactionsPage) Layout(gtx layout.Context) layout.Dimensions {
+func (pg *TransactionsPage) Layout(gtx C) D {
 	if pg.Load.GetCurrentAppWidth() <= gtx.Dp(values.StartMobileView) {
 		return pg.layoutMobile(gtx)
 	}
 	return pg.layoutDesktop(gtx)
 }
 
-func (pg *TransactionsPage) layoutDesktop(gtx layout.Context) layout.Dimensions {
+func (pg *TransactionsPage) walletNotReady() bool {
+	return pg.selectedWallet != nil && (!pg.selectedWallet.IsSynced() || pg.selectedWallet.IsRescanning())
```

Comment:
I think wallet is not ready if it is no wallet is selected as well. Am I correct?
So basically, wallet is not ready if selected wallet is nil or it's not synced or it's rescanning.
```suggestion
	return pg.selectedWallet == nil || !pg.selectedWallet.IsSynced() || pg.selectedWallet.IsRescanning()
```

Reply:
You are not correct.. the fact that a wallet is not selected does not mean we should not show txs of wallet that are synced.

When fetching txs for multi wallets, we check the sync status.. if it not synced, we skip that wallet.